runners/samza/src/main/java/org/apache/beam/runners/samza/SamzaRunner.java

```diff
@@ -106,6 +107,7 @@ public PortablePipelineResult runPortablePipeline(RunnerApi.Pipeline pipeline) {
 
   @Override
   public SamzaPipelineResult run(Pipeline pipeline) {
+    SplittableParDo.validateNoPrimitiveReads(pipeline);
```

Comment:
is it expected that we only validate in non-portable mode? i.e. this validation does not exist in runPortablePipeline()

Reply:
Yes since portable pipelines only support SDF. This is about migrating non-portable pipeline runner implementations to use SDF.